### PR TITLE
Bhavik/add delete for upload

### PR DESCRIPTION
**Pull request by br0wncrust4cean**
- Added ability to delete a file for the uploader

### CODE REVIEW

**Review by acournoyer1 — APPROVED** [review]

File: Project/pom.xml [hunk 1]
```diff
@@ -35,6 +35,13 @@
...
             <version>1.9.1</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.webjars/font-awesome -->
+        <dependency>
+            <groupId>org.webjars</groupId>
+            <artifactId>font-awesome</artifactId>
+            <version>4.7.0</version>
+        </dependency>
+
         <dependency>
```
Comment (Taz-Z) [inline Project/pom.xml:45]: https://mvnrepository.com/artifact/org.webjars/font-awesome/5.7.2, dont use old version, you an upto date one please

File: Project/src/main/java/app/FileUploadController.java [hunk 2]
```diff
@@ -74,4 +71,15 @@ public String uploadFile(@RequestParam("file") MultipartFile file) throws IOExce
...
     }
 
+    /**
+     * Deletes an article from the article repository
+     * @param id The id of the article to be deleted
+     * @return
+     */
+    @DeleteMapping("/upload/{id}")
+    public ResponseEntity<Void> deleteFileById(@PathVariable(value = "id") Long id) {
+        if (id <= 0 || !articleRepository.existsById(id)) throw new BadFileException();
```
Comment (Taz-Z) [inline Project/src/main/java/app/FileUploadController.java]: Isnt id <=0 useless check? articleRepository.existsById(id) should handle zero and negative numbers as well

File: Project/src/main/resources/static/js/upload.js [hunk 3]
```diff
@@ -30,6 +30,25 @@ function alertUpload(errorMsg) {
...
+            contentType: "application/json",
+            headers: {
+                'X-CSRF-TOKEN': $('[name="_csrf"]').attr('value')
+            },
+            success: function () {
+                $("#article" + id).remove();
+                $("#successAlert").show();
+            }
+        });
+    } else {
```
Comment (Taz-Z) [inline Project/src/main/resources/static/js/upload.js]: no need for else, just return

File: Project/src/main/resources/templates/upload.html [hunk 4]
```diff
@@ -46,16 +51,21 @@
                     <th>Upload Date</th>
                     <th>Reviewer</th>
                     <th>State</th>
-                    <th>Download</th>
+                    <th></th>
```
Comment (Taz-Z) [inline Project/src/main/resources/templates/upload.html]: no header, Download or something would be fine

**Review by Taz-Z — CHANGES_REQUESTED** [review]

File: Project/pom.xml (hunk 1 above)
Comment (br0wncrust4cean) [reply Project/pom.xml:45]: Latest version didn't work. Literally the logo just didn't show up. This one was the latest I could use.

Comment (br0wncrust4cean) [reply Project/pom.xml:45]: If we wanted to use latest versions as well, then we have outdated versions of jquery and bootstrap imported in here.

Comment (Taz-Z) [reply Project/pom.xml:45]: Ya we should grab the latest, font

https://fontawesome.com/icons/trash-alt?style=solid

youre using fa as your class prefix, thats old

far/fas are new 

<i class="fas fa-trash-alt"></i>

**Review by Taz-Z — APPROVED** [review]

**Pull request merged**